src/server/examples/examples_test.go

```diff
@@ -73,3 +73,184 @@ func TestExampleTensorFlow(t *testing.T) {
 	require.NoError(t, c.DeleteRepo("GoT_train", false))
 	require.NoError(t, c.DeleteRepo("GoT_scripts", false))
 }
+
+func TestWordCount(t *testing.T) {
+
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c := getPachClient(t)
+
+	// Should stay in sync with doc/examples/word_count/README.md
+	inputPipelineManifest := `
+{
+  "pipeline": {
+    "name": "wordcount_input"
+  },
+  "transform": {
+    "image": "pachyderm/job-shim:latest",
+    "cmd": [ "wget",
+        "-e", "robots=off",
+        "--recursive",
+        "--level", "1",
+        "--adjust-extension",
+        "--no-check-certificate",
+        "--no-directories",
+        "--directory-prefix",
+        "/pfs/out",
+        "https://news.ycombinator.com/newsfaq.html"
+    ],
+    "acceptReturnCode": [4,5,6,7,8]
+  },
+  "parallelism_spec": {
+       "strategy" : "CONSTANT",
+       "constant" : 1
+  }
+}
+`
+	exampleDir := "../../../doc/examples/word_count"
+	cmd := exec.Command("pachctl", "create-pipeline")
+	cmd.Stdin = strings.NewReader(inputPipelineManifest)
+	cmd.Dir = exampleDir
+	_, err := cmd.CombinedOutput()
+	require.NoError(t, err)
+
+	cmd = exec.Command("pachctl", "run-pipeline", "wordcount_input")
+	cmd.Dir = exampleDir
+	_, err = cmd.Output()
+	require.NoError(t, err)
+
+	cmd = exec.Command("docker", "build", "-t", "wordcount-map", ".")
+	cmd.Dir = exampleDir
+	_, err = cmd.CombinedOutput()
+	require.NoError(t, err)
+
+	// Should stay in sync with doc/examples/word_count/README.md
+	wordcountMapPipelineManifest := `
+{
+  "pipeline": {
+    "name": "wordcount_map"
+  },
+  "transform": {
+    "image": "wordcount-map:latest",
+    "cmd": ["/map", "/pfs/wordcount_input", "/pfs/out"]
+  },
+  "inputs": [
+    {
+      "repo": {
+        "name": "wordcount_input"
+      }
+    }
+  ]
+}
+`
+	cmd = exec.Command("pachctl", "create-pipeline")
+	cmd.Stdin = strings.NewReader(wordcountMapPipelineManifest)
+	cmd.Dir = exampleDir
+	_, err = cmd.Output()
+	require.NoError(t, err)
+
+	// Flush Commit can't help us here since there are no inputs
+	// So we poll wordcount_input until it has a commit
```

Comment:
Setting the last parameter on `ListCommit` to `true` will make it block until there's a result. Probably easier than a loop.